enhancements/microshift/microshift-apiserver-certs.md

```diff
@@ -0,0 +1,312 @@
+---
+title: microshift-apiserver-certs
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@fzdarsky, MicroShift architect"
+  - "@ggiguash, MicroShift contributor"
+  - "@stlaz, Security specialist"
+  - "@pmtk, MicroShift contributor"
+  - "@copejon, MicroShift contributor"
+  - "@deads2k, OpenShift architect"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - N/A
+creation-date: 2023-01-20
+last-updated: 2023-01-20
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-716
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# MicroShift API server certificates
+## Summary
+This enhancement proposes a strategy to manage API server certificates in all
+kinds of environments, both externally (accessible through kubeconfig) and
+internally (pods reaching the API server).
+
+MicroShift has a wide variety of different use cases when it comes to API
+server connectivity, we can find environments where the cluster needs to be
+reachable with the IP, with the hostname, by some other name having DNS, or
+a combination of them.
+
+## Motivation
+MicroShift does not cover all requirements and use cases right now. It shares
+certificate handling with OCP, which is a bit different in terms of deployment.
+
+MicroShift does not have load balancers before the API server, and is required
+to be reachable and authenticate by using the IP, hostname, a DNS name, or a
+combination of those. In order to fulfill industrial use cases, we propose
+having a single API server certificate.
```

Comment:
I think we want a clear problem summary here. The paragraph you have explains the requirement, and the rest of the text below gives the details, but we should set the context with a couple of lines summarizing the problem. Something like:

Unlike OCP, MicroShift users need to be able to access the API server via IP address as well as hostname. Therefore, the IP needs to be in the identities (CN or SAN) for the certificate. Because of the way certificate lookup works in the API server, there can only be one certificate associated with an IP address. Because of the network architecture of MicroShift, external clients and internal clients appear to use the same server IP. Therefore, connections from both would use the same server certificate to authenticate the connection.